pallets/rmrk-core/src/functions.rs

```diff
@@ -1,5 +1,69 @@
 use super::*;
 
+impl<T: Config> Resource<StringLimitOf<T>, T::AccountId> for Pallet<T> {
+	fn resource_add(
+		sender: T::AccountId,
+		collection_id: CollectionId,
+		nft_id: NftId,
+		base: Option<BoundedVec<u8, T::StringLimit>>,
+		src: Option<BoundedVec<u8, T::StringLimit>>,
+		metadata: Option<BoundedVec<u8, T::StringLimit>>,
+		slot: Option<BoundedVec<u8, T::StringLimit>>,
+		license: Option<BoundedVec<u8, T::StringLimit>>,
+		thumb: Option<BoundedVec<u8, T::StringLimit>>,
+	) -> Result<ResourceId, DispatchError> {
+		let mut pending = false;
+		let nft = NFTs::<T>::get(collection_id, nft_id).ok_or(Error::<T>::NoAvailableNftId)?;
+		if nft.rootowner != sender {
+			pending = true;
+		}
+
+		let resource_id = Self::get_next_resource_id()?;
+		ensure!(
+			Resources::<T>::get((collection_id, nft_id, resource_id)).is_none(),
+			Error::<T>::ResourceAlreadyExists
+		);
+
+		let empty = base.is_none()
+			&& src.is_none()
+			&& metadata.is_none()
+			&& slot.is_none()
+			&& license.is_none()
+			&& thumb.is_none();
+		ensure!(!empty, Error::<T>::EmptyResource);
+
+		let res =
+			ResourceInfo { id: resource_id, base, src, metadata, slot, license, thumb, pending };
```

Comment:
consider simplifying this bit:
```
pending: nft.rootowner != sender,
```
so no need for extra `pending` mut variable

Reply:
https://github.com/rmrk-team/rmrk-substrate/pull/41/commits/cc295151e72e9728c6e3e0c1f22747c3070a3c0b